github/util.go

```diff
@@ -105,3 +106,12 @@ func validateTeamIDFunc(v interface{}, keyName string) (we []string, errors []er
 
 	return
 }
+
+func prepareResourceContext(d *schema.ResourceData) context.Context {
+	ctx := context.WithValue(context.Background(), ctxId, d.Id())
+	if etag, set := d.GetOk(ctxEtag); set {
+		ctx = context.WithValue(ctx, ctxEtag, etag.(string))
+	}
```

Comment:
This seems safe but introduces a potential risk if not equivalent to the replaced block:
```
if !d.IsNewResource() {	
	ctx = context.WithValue(ctx, ctxEtag, d.Get("etag").(string))	
}
```

This doesn't block my approval but figured I'd call it out in case anyone else more experienced has input.